src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -304,6 +308,15 @@ public override Task<T> WaitForExternalEvent<T>(string name)
                     reason: $"WaitForExternalEvent:{name}",
                     isReplay: this.innerContext.IsReplaying);
 
+                // Check the queue to see if any events came in before the orchestrator was listening
+                if (this.bufferedExternalEvents.TryGetValue(name, out Queue queue))
+                {
+                    object input = queue.Dequeue();

```

Comment:
If the queue is empty after dequeuing, I wonder if we should consider removing it from the `bufferedExternalEvents` dictionary. My worry is that someone could write an eternal orchestration that listens for dynamically named events, and their instance could run out of memory. It might not be very realistic, but I don't think it would hurt us to clean these up.